src/models/Order.js

```diff
@@ -137,6 +134,30 @@ OrderSchema.pre('findOne', async function (next) {
   console.log('Pre-findOne finished');
   next();
 });
+
+//AKOS: this post middleware is not being triggered after updating order status (node updateOrderToPaidScript numberOfOrder)
+// send order completion email when the order status changes to "payment_successful"
+OrderSchema.post('findByIdAndUpdate', async function (doc) {
```

Comment:
Looks like the correct middleware name is actually `findOneAndUpdate` (it should be triggered also by calls to `findByIdAndUpdate` because under-the-hood Mongoose will actually just be calling `findOneAndUpdate` for that function anyway): `https://mongoosejs.com/docs/api/query.html#Query.prototype.findOneAndUpdate()`

Reply:
In Mongoose, 
The findOneAndUpdate function is used to find a single document that matches a specified condition and update it. It returns the updated document.
findByIdAndUpdate:

The findByIdAndUpdate function, on the other hand, is used to find a document by its unique identifier (usually the _id field) and update it. It also returns the updated document.
//here we set a condition ourselves  `if (this.getUpdate().$set.orderStatus === 'payment_successful')`, correct middleware name is `findOneAndUpdate`

Reply:
Yes; to clarify, in the code we'd still call `findByIdAndUpdate` but the middleware must be called `findOneAndUpdate`: [https://stackoverflow.com/a/73215258](https://stackoverflow.com/a/73215258)  
 

> `findByIdAndUpdate()` is essentially a wrapper for `findOneAndUpdate()`. That is, it is _equivalent to_ `findOneAndUpdate({ _id: id }, ...)`.
> 
> As such, it **does not bypasses** mongoose middleware completely.
> 
> As per the [documentation](https://mongoosejs.com/docs/api.html#model_Model-findByIdAndUpdate):
> 
> > This function triggers the following middleware.
> > 
> > *   `findOneAndUpdate()`

Reply:
@akosasante thank you! 
link to docs: https://mongoosejs.com/docs/api/model.html#Model.findByIdAndUpdate()